libs/wingc/src/jsify.rs

```diff
@@ -532,6 +532,9 @@ impl<'a> JSifier<'a> {
 					context,
 				),
 			},
+			ExprKind::OptionalTest { optional } => {
+				format!("(({}) !== null)", self.jsify_expression(optional, context))
```

Comment:
So this is an interesting question, do we want to check against `null` or against `undefined`, or against both? (These are distinct values in JavaScript because, hey why not 🙃).

My slight inclination would be to check against `undefined` OR to have our language ignore the distinction between `null` and `undefined` in JS and just treat them the same. I'll list some thoughts below but there isn't really a clear/best option to me so I'm also fine leaving this as is.

```suggestion
				format!("((val) => val !== null && val !== undefined)({})", self.jsify_expression(optional, context))
```

Thoughts:
1. When function arguments are omitted in JS, they are treated as `undefined` when you try accessing them, which works out for us
2. JSII libraries don't distinguish between `null` and `undefined` (https://aws.github.io/jsii/specification/2-type-system/#null-and-undefined)
3. More flexible if someone wants to write an API in JS that uses optionals, as their API can return `null` or `undefined` and both are fine. See the example below:

```ts
// helpers.ts
export function func1(): string | undefined {
  return Math.random() > 0.5 ? "hello" : undefined;
}
export function func2(): string | null {
  return Math.random() > 0.5 ? "hello" : null;
}

// wing
resource Utils {
  // do we care to treat these differently? probably not
  extern "./helpers.ts" static inflight func1(): str?;
  extern "./helpers.ts" static inflight func2(): str?;
}
```

Reply:
For JSII compatibility, I would consider both JS `null` and `undefined` as `false`.

Reply:
ok